lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -21,11 +47,17 @@ def initialize(serializer, options = {})
         def serializable_hash(options = nil)
           options ||= {}
 
-          if serializer.respond_to?(:each)
-            serializable_hash_for_collection(options)
-          else
-            serializable_hash_for_single_resource(options)
-          end
+          hash =
+            case
```

Comment:
When I was a young dev I learned to dislike using the return value from and if/else, since it's not as comfortable as using the return value from a case/switch